tests/observability/suite.go

```diff
@@ -193,22 +195,29 @@ func testPodDisruptionBudgets(check *checksdb.Check, env *provider.TestEnvironme
 		check.LogInfo("Testing Deployment %q", d.ToString())
 		pdbFound := false
 		for pdbIndex := range env.PodDisruptionBudgets {
-			for k, v := range d.Spec.Template.Labels {
-				if env.PodDisruptionBudgets[pdbIndex].Spec.Selector.MatchLabels[k] == v {
-					pdbFound = true
-					if ok, err := pdbv1.CheckPDBIsValid(&env.PodDisruptionBudgets[pdbIndex], d.Spec.Replicas); !ok {
-						check.LogError("PDB %q is not valid for Deployment %q, err: %v", env.PodDisruptionBudgets[pdbIndex].Name, d.Name, err)
-						nonCompliantObjects = append(nonCompliantObjects, testhelper.NewReportObject(fmt.Sprintf("Invalid PodDisruptionBudget config: %v", err), testhelper.DeploymentType, false).
-							AddField(testhelper.DeploymentName, d.Name).
-							AddField(testhelper.Namespace, d.Namespace).
-							AddField(testhelper.PodDisruptionBudgetReference, env.PodDisruptionBudgets[pdbIndex].Name))
-					} else {
-						check.LogInfo("PDB %q is valid for Deployment: %q", env.PodDisruptionBudgets[pdbIndex].Name, d.Name)
-						compliantObjects = append(compliantObjects, testhelper.NewReportObject("Deployment: references PodDisruptionBudget", testhelper.DeploymentType, true).
-							AddField(testhelper.DeploymentName, d.Name).
-							AddField(testhelper.Namespace, d.Namespace).
-							AddField(testhelper.PodDisruptionBudgetReference, env.PodDisruptionBudgets[pdbIndex].Name))
-					}
+			pdb := &env.PodDisruptionBudgets[pdbIndex]
+			if pdb.Namespace != d.Namespace {
+				continue
+			}
+			pdbSelectorMap, err := metav1.LabelSelectorAsMap(pdb.Spec.Selector)
```

Comment:
`LabelSelectorAsMap` never used that one before.  Nice!

Reply:
@jmontesi by "Also, take into account the "MatchExpressions" field of the selector when it can be properly converted to a map." do you mean to leave out some MatchExpressions that are not covered by LabelSelectorAsMap ( e.g. 
"Operators that cannot be converted into plain labels (Exists, DoesNotExist, NotIn, and In with more than one value") ? Shouldn't we support all these cases?

Reply:
Yes, I left those cases out as I considered them quite unlikely to be used in practice to match a PDB with a workload. But reviewing again the Kubernetes API I found a way to nicely cover all cases.